In [1]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [2]:
# Importing Libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import plotly.express as px

import warnings
warnings.filterwarnings("ignore")
warnings.simplefilter(action='ignore', category=FutureWarning)

In [3]:
data=pd.read_csv('/content/drive/MyDrive/Colab Notebooks/Data/Iris.csv',encoding="ISO-8859-1") #reading csv file using pandas
print(data.head())

   Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm      Species
0   1            5.1           3.5            1.4           0.2  Iris-setosa
1   2            4.9           3.0            1.4           0.2  Iris-setosa
2   3            4.7           3.2            1.3           0.2  Iris-setosa
3   4            4.6           3.1            1.5           0.2  Iris-setosa
4   5            5.0           3.6            1.4           0.2  Iris-setosa


In [4]:
print(data.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Id             150 non-null    int64  
 1   SepalLengthCm  150 non-null    float64
 2   SepalWidthCm   150 non-null    float64
 3   PetalLengthCm  150 non-null    float64
 4   PetalWidthCm   150 non-null    float64
 5   Species        150 non-null    object 
dtypes: float64(4), int64(1), object(1)
memory usage: 7.2+ KB
None


Information about data set is:

*   150 rows
*   6 columns
*   1 columns have object data type
*   5 columns have float data type

In [5]:
print(data.describe())

               Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm
count  150.000000     150.000000    150.000000     150.000000    150.000000
mean    75.500000       5.843333      3.054000       3.758667      1.198667
std     43.445368       0.828066      0.433594       1.764420      0.763161
min      1.000000       4.300000      2.000000       1.000000      0.100000
25%     38.250000       5.100000      2.800000       1.600000      0.300000
50%     75.500000       5.800000      3.000000       4.350000      1.300000
75%    112.750000       6.400000      3.300000       5.100000      1.800000
max    150.000000       7.900000      4.400000       6.900000      2.500000


In [6]:
# Checking description of dtype objects
data.describe(include= "object")

Species
count           150
unique            3
top     Iris-setosa
freq             50

In [7]:
data['Species'].unique()

array(['Iris-setosa', 'Iris-versicolor', 'Iris-virginica'], dtype=object)

In [8]:
data.groupby('Species')[['SepalLengthCm', 'SepalWidthCm']].describe()

SepalLengthCm                                              \
                        count   mean       std  min    25%  50%  75%  max   
Species                                                                     
Iris-setosa              50.0  5.006  0.352490  4.3  4.800  5.0  5.2  5.8   
Iris-versicolor          50.0  5.936  0.516171  4.9  5.600  5.9  6.3  7.0   
Iris-virginica           50.0  6.588  0.635880  4.9  6.225  6.5  6.9  7.9   

                SepalWidthCm                                                
                       count   mean       std  min    25%  50%    75%  max  
Species                                                                     
Iris-setosa             50.0  3.418  0.381024  2.3  3.125  3.4  3.675  4.4  
Iris-versicolor         50.0  2.770  0.313798  2.0  2.525  2.8  3.000  3.4  
Iris-virginica          50.0  2.974  0.322497  2.2  2.800  3.0  3.175  3.8

In [9]:
data.groupby('Species')[['PetalLengthCm','PetalWidthCm']].describe()

PetalLengthCm                                               \
                        count   mean       std  min  25%   50%    75%  max   
Species                                                                      
Iris-setosa              50.0  1.464  0.173511  1.0  1.4  1.50  1.575  1.9   
Iris-versicolor          50.0  4.260  0.469911  3.0  4.0  4.35  4.600  5.1   
Iris-virginica           50.0  5.552  0.551895  4.5  5.1  5.55  5.875  6.9   

                PetalWidthCm                                            
                       count   mean       std  min  25%  50%  75%  max  
Species                                                                 
Iris-setosa             50.0  0.244  0.107210  0.1  0.2  0.2  0.3  0.6  
Iris-versicolor         50.0  1.326  0.197753  1.0  1.2  1.3  1.5  1.8  
Iris-virginica          50.0  2.026  0.274650  1.4  1.8  2.0  2.3  2.5

In [10]:
data.groupby('Species')[['SepalLengthCm', 'SepalWidthCm']].sum()/50

SepalLengthCm  SepalWidthCm
Species                                     
Iris-setosa              5.006         3.418
Iris-versicolor          5.936         2.770
Iris-virginica           6.588         2.974

In [11]:
data.groupby('Species')[['PetalLengthCm','PetalWidthCm']].sum()/50

PetalLengthCm  PetalWidthCm
Species                                     
Iris-setosa              1.464         0.244
Iris-versicolor          4.260         1.326
Iris-virginica           5.552         2.026

In [17]:
data[data['Species']=='Iris-setosa']

Id  SepalLengthCm  SepalWidthCm  PetalLengthCm  PetalWidthCm      Species
0    1            5.1           3.5            1.4           0.2  Iris-setosa
1    2            4.9           3.0            1.4           0.2  Iris-setosa
2    3            4.7           3.2            1.3           0.2  Iris-setosa
3    4            4.6           3.1            1.5           0.2  Iris-setosa
4    5            5.0           3.6            1.4           0.2  Iris-setosa
5    6            5.4           3.9            1.7           0.4  Iris-setosa
6    7            4.6           3.4            1.4           0.3  Iris-setosa
7    8            5.0           3.4            1.5           0.2  Iris-setosa
8    9            4.4           2.9            1.4           0.2  Iris-setosa
9   10            4.9           3.1            1.5           0.1  Iris-setosa
10  11            5.4           3.7            1.5           0.2  Iris-setosa
11  12            4.8           3.4            1.6           0.2  Iris-setosa
12  13            4.8           3.0            1.4           0.1  Iris-setosa
13  14            4.3           3.0            1.1           0.1  Iris-setosa
14  15            5.8           4.0            1.2           0.2  Iris-setosa
15  16            5.7           4.4            1.5           0.4  Iris-setosa
16  17            5.4           3.9            1.3           0.4  Iris-setosa
17  18            5.1           3.5            1.4           0.3  Iris-setosa
18  19            5.7           3.8            1.7           0.3  Iris-setosa
19  20            5.1           3.8            1.5           0.3  Iris-setosa
20  21            5.4           3.4            1.7           0.2  Iris-setosa
21  22            5.1           3.7            1.5           0.4  Iris-setosa
22  23            4.6           3.6            1.0           0.2  Iris-setosa
23  24            5.1           3.3            1.7           0.5  Iris-setosa
24  25            4.8           3.4            1.9           0.2  Iris-setosa
25  26            5.0           3.0            1.6           0.2  Iris-setosa
26  27            5.0           3.4            1.6           0.4  Iris-setosa
27  28            5.2           3.5            1.5           0.2  Iris-setosa
28  29            5.2           3.4            1.4           0.2  Iris-setosa
29  30            4.7           3.2            1.6           0.2  Iris-setosa
30  31            4.8           3.1            1.6           0.2  Iris-setosa
31  32            5.4           3.4            1.5           0.4  Iris-setosa
32  33            5.2           4.1            1.5           0.1  Iris-setosa
33  34            5.5           4.2            1.4           0.2  Iris-setosa
34  35            4.9           3.1            1.5           0.1  Iris-setosa
35  36            5.0           3.2            1.2           0.2  Iris-setosa
36  37            5.5           3.5            1.3           0.2  Iris-setosa
37  38            4.9           3.1            1.5           0.1  Iris-setosa
38  39            4.4           3.0            1.3           0.2  Iris-setosa
39  40            5.1           3.4            1.5           0.2  Iris-setosa
40  41            5.0           3.5            1.3           0.3  Iris-setosa
41  42            4.5           2.3            1.3           0.3  Iris-setosa
42  43            4.4           3.2            1.3           0.2  Iris-setosa
43  44            5.0           3.5            1.6           0.6  Iris-setosa
44  45            5.1           3.8            1.9           0.4  Iris-setosa
45  46            4.8           3.0            1.4           0.3  Iris-setosa
46  47            5.1           3.8            1.6           0.2  Iris-setosa
47  48            4.6           3.2            1.4           0.2  Iris-setosa
48  49            5.3           3.7            1.5           0.2  Iris-setosa
49  50            5.0           3.3            1.4           0.2  Iris-setosa

In [12]:
(px.box(data['SepalLengthCm'][data['Species']=='Iris-setosa']))

In [13]:
(px.box(data['SepalWidthCm'][data['Species']=='Iris-setosa']))

In [14]:
(px.box(data['PetalLengthCm'][data['Species']=='Iris-setosa']))

In [15]:
(px.box(data['PetalWidthCm'][data['Species']=='Iris-setosa']))